### ISSUE TITLE
Cascaded Shadow Maps crashes entire scene on iOS

### ISSUE THREAD

**Issue by Awendel** [BabylonJS/Babylon.js#7813]
Verified using both iPad & iPhone:

Cascaded Shadow Maps lead to Script Error / crash of entire Scene.

Problems does not occur on Android, likely due to it supporting WebGL2.

If the feature can't be implemented on iOS, it should instead provide a warning and not crash the scene (ideally of course it would fallback to normal shadow Generator, but I suppose this logic could also be implemented by the developer).

Applies to all Playground examples that are references on the following page:

https://doc.babylonjs.com/babylon101/shadows_csm

**Comment by sebavan** [comment]
Totally agree 👍

**Comment by sebavan** [comment]
@Popov72 could you check this one ?

**Comment by Popov72** [comment]
Well, it's already handled.

In the CSM constructor:

```typescript
        if (this._scene.getEngine().webGLVersion == 1) {
            throw "CSM can only be used in WebGL2";
        }
```
it's up to the developer to catch the exception and handle it gracefully.

**Comment by Popov72** [comment]
I don't think we can do any better as we are in a constructor, we can't return a specific value to indicate it is not supported.

**Comment by sebavan** [comment]
Ohhhhhh I thought there were no custom message, my bad. Do you think we should log a message instead ? as we do in other places in babylon ?

**Comment by sebavan** [comment]
A bit like we do in SSA02RenderingPipeline with a isSupported method and an error log ???

**Comment by Popov72** [comment]
Ok, going to make a PR for that!

**Comment by Popov72** [comment]
See #7817

**Comment by sebavan** [comment]
Closed by #7817
